dags/utils/gcp.py

```diff
@@ -700,3 +714,65 @@ def gke_command(
         env_vars=context_env_vars,
         **kwargs
     )
+
+
+def publish_json(
+    destination_table,
+    dataset_id,
+    parameters=(),
+    arguments=(),
+    project_id=None,
+    sql_file_path=None,
+    target_bucket=None,     # todo: once available, set name here
+    gcp_conn_id="google_cloud_derived_datasets",
+    gke_location="us-central1-a",
+    gke_cluster_name="bq-load-gke-1",
+    gke_namespace="default",
+    docker_image="mozilla/bigquery-etl:latest",
+    date_partition_parameter="submission_date",
+    **kwargs
+):
+    """
+    Publish query results as JSON to public Cloud Storage bucket.
+
+    :param str destination_table:                  [Required] BigQuery destination table
+    :param str dataset_id:                         [Required] BigQuery default dataset id
+    :param Tuple[str] parameters:                  Parameters passed to bq query
+    :param Tuple[str] arguments:                   Additional bq query arguments
+    :param Optional[str] project_id:               BigQuery default project id
+    :param Optional[str] sql_file_path:            Optional override for path to the
+                                                   SQL query file to run
+    :param str gcp_conn_id:                        Airflow connection id for GCP access
+    :param str gke_location:                       GKE cluster location
+    :param str gke_cluster_name:                   GKE cluster name
+    :param str gke_namespace:                      GKE cluster namespace
+    :param str docker_image:                       docker image to use
+    :param Optional[str] date_partition_parameter: Parameter for indicating destination
+                                                   partition to generate, if None
+                                                   destination should be whole table
+                                                   rather than partition
+    :param Dict[str, Any] kwargs:                  Additional keyword arguments for
+                                                   GKEPodOperator
+    """
+
+    kwargs["task_id"] = "publish_json_" + destination_table
+    kwargs["name"] = kwargs["task_id"].replace("_", "-")[:63]
```

Comment:
We need to ensure that name is <64 characters long. By prepending `publish_json_` some tests failed because the name was too long.

Reply:
Can you maybe add a comment here, and make `64` a constant?